doc/design/OrionHDFSSink.md

```diff
@@ -260,12 +262,14 @@ There exists an [issue](https://github.com/telefonicaid/fiware-cosmos/issues/111
 
 [Top](#top)
 
-####<a name="section2.3.3"></a>About the batch size
+####<a name="section2.3.3"></a>About batching
 As seen in the [implementation section](#section4), `OrionHDFSSink` extends `OrionSink`, which provides a built-in mechanism for collecting events from the internal Flume channel. This mechanism allows exteding classes have only to deal with the persistence details of such a batch of events in the final backend.
 
 What is important regarding the batch mechanism is it largely increases the performance of the sink, because the number of writes is dramatically reduced. Let's see an example, let's assume a batch of 100 Flume events. In the best case, all these events regard to the same entity, which means all the data within them will be persisted in the same HDFS file. If processing the events one by one, we would need 100 writes to HDFS; nevertheless, in this example only one write is required. Obviously, not all the events will always regard to the same unique entity, and many entities may be involved within a batch. But that's not a problem, since several sub-batches of events are created within a batch, one sub-batch per final destination HDFS file. In the worst case, the whole 100 entities will be about 100 different entities (100 different HFS destinations), but that will not be the usual scenario. Thus, assuming a realistic number of 10-15 sub-batches per batch, we are replacing the 100 writes of the event by event approach with only 10-15 writes.
 
-By default, `OrionHDFSSink` has a configured batch size of one, but as explained above, it is highly recommended to increase the batch size for performance purposes. Which is the optimal value? The size of the batch it is closely related to the transaction size of the channel the events are got from (it has no sense the first one is greater then the second one), and it depends on the number of estimated sub-batches as well. A deeper discussion on the batches of events and their appropriate sizing may be found in the [performance document](../operation/performance_tuning_tips.md).
+The batch mechanism adds an accumlation timeout to protect the sink to stay in an eternal state of batch building when no new data arrives. If such a timeout is reached, then the batch is persisted as it is.
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/048eecb2843567c6788449fb8c10bbc6773450e5